starter-code/js/index.js

```diff
@@ -1,52 +1,116 @@
+    // console.log("res :" + totalAmount);
+    e.innerHTML = "$ " + totalPrice[idx];
+  });
+
+  // TOTAL CART AMOUNT
+  var totalCartAmount = document.querySelector(".total-price-span");
+  totalCartAmount.innerHTML = totalAmount;
+
+  // SINGLE LOOP TRAIL
+  // var totalPriceHolder = document.querySelectorAll(".product-total-price");
+  // totalPriceHolder.forEach(function(e,idx){
+  // });
 }
 
-function createQuantityInput(){
+function createQuantityInput() {}
 
-}
+function createDeleteButton() {}
 
-function createDeleteButton(){
+function createQuantityNode() {}
 
-}
+function createItemNode(dataType, itemData) {}
 
-function createQuantityNode(){
+function createNewItemRow(itemName, itemUnitPrice) {}
 
-}
-
-function createItemNode(dataType, itemData){
-
-}
-
-function createNewItemRow(itemName, itemUnitPrice){
+function createNewItem() {
+  // Selector
+  var addProduct = document.querySelector("#new-item-create");
+  console.log(addProduct);
+  
+  addProduct.onclick = function(){
+    console.log("Add product clicked");
+    
+    var inputName = document.querySelector(".input-product-name");
+    var inputNameValue = document.querySelector(".input-product-name").value;
+    inputName.classList.add("product-name");
+    var inputPrice = document.querySelector(".input-product-price");
+    var inputPriceValue = document.querySelector(".input-product-price").value;
+    inputPrice.classList.add("product-price");
+    var inputQuantity = "<div class='quantity'>quantity<input type='text' class='quantity-input' placeholder='enter your quantity'/> </div>"
+    var delBtn = "<div><button class='btn btn-delete'>DELETE</button></div>"
+    var newLine = document.createElement("div");
+    newLine.classList.add("container");
+    newLine.innerHTML = "<div class='product-name'><span>" + inputNameValue + "</span></div>" + "<div class='product-price'>$<span class='product-price-span'>"+inputPriceValue+"</span></div>" + inputQuantity + "<div class='product-total-price'>0.00</div>" + delBtn;
 
-}
+    var list = document.querySelector(".global-container");
+    list.appendChild(newLine);
+  };
 
-function createNewItem(){
 
 }
 
-window.onload = function(){
-  var calculatePriceButton = document.getElementById('calc-prices-button');
+window.onload = function() {
+  var defaultPrice = document.querySelectorAll(".product-total-price");
+  defaultPrice.forEach(function(e, idx) {
+    e.innerHTML = "0.00";
+  });
+  var calculatePriceButton = document.getElementById("calc-prices-button");
   var createItemButton = document.getElementById('new-item-create');
-  var deleteButtons = document.getElementsByClassName('btn-delete');
+  var deleteButtons = document.getElementsByClassName("btn-delete");
 
   calculatePriceButton.onclick = getTotalPrice;
   createItemButton.onclick = createNewItem;
 
-  for(var i = 0; i<deleteButtons.length ; i++){
+  for (var i = 0; i < deleteButtons.length; i++) {
     deleteButtons[i].onclick = deleteItem;
   }
 };
```

Comment:
You put your delete button in an onclick event so that's why it's working only when you are clicking first on a delete button that is already existing. If you create a new item after having deleted the two first existing lines, it's impossible to delete it any more. 

Here is a solution to add an item and make the delete button working: 
```
  // Defining the function, will be called upon clicking on a button
  function createItem() {
    
    // I created a wrapper in the HTML, so I can appendChild the divs I'm going to create to it
    var itemsList = document.querySelector( ".items-wrapper" );

    // I create a new "div" DOM element, and save it inside my newDiv variable
    var newDiv = document.createElement( "div" );

    // I want my div to have the "wrapper" class, so I use setAttribute
    newDiv.setAttribute( "class", "wrapper" );

    // I created two inputs in the HTML, so the user can choose a name and a price for the new product
    // I gave those inputs IDs, and now I save them inside those variable to be able to retrieve their values
    var itemName = document.querySelector( "#name" );
    var itemPrice = document.querySelector( "#price" );
    
    // I want my newDiv to have the same HTML structure than the already existing items, so I create a content variable with it
    // Here I didn't use regular "" or '', but `` (backticks, or accent grave).
    // Those allow two cool things:
    //    I can break lines, and thus have a proper HTML structure with indentation: easy to write and to read
    //    I can easily use javascript variables inside it, with the ${ myVariable } syntax
    // having those variable that hold my inputs, I can easily inject them inside my content's HTML structure
    var content = `
      <div><span>${ itemName.value }</span></div>
      
      <div>$<span class="unit-price" id="unit-price">${ itemPrice.value }</span></div>
      
      <div>
        <label for="units" class="quantity">QTY</label>
        <input class="unit-quantity" id="units" placeholder="Buy ALL the things!">
      </div>
      
      <div>$<span class="total-price" id="total-price">0.00</span></div>
      
      <div>
        <button class="btn btn-delete">Delete</button>
      </div>
    `;

    // Now that I have have my content, I can simply add it inside my newDiv, using innerHTML
    newDiv.innerHTML = content;

    // My newDiv is ready, I can add it to the actual DOM using appendChild
    itemsList.appendChild( newDiv );


    // Once I'm done with creating the new item line, I clear the inputs by redefining their values to empty string ""
    itemName.value = "";
    itemPrice.value = "";
  
    // I created my new item line along with the delete button.
    // But since the delete buttons collection is created earlier, before I create my new line, my new delete button doesn't have the onclick function
    // Thus, I need to make sure it's inside the delete buttons collection by updating the deleteButtons value, like so:
    deleteButtons = document.querySelectorAll(".btn-delete");

    // And then, I need to apply the onclick function again on all the buttons of the collection,
    // so I'm sure the newly created button has the onclick function too
    deleteButtons.forEach( each => {
      each.onclick = deleteItem;
    })
  }
    
    // Here I assign my HTML button to a Javascript variable
    var createButton = document.querySelector( ".create-button" );

    // And I assign the createItem I just defined to it
    createButton.onclick = createItem;
```